src/lib/y2users/linux/reader.rb

```diff
@@ -96,25 +99,36 @@ def read_groups(config)
 
       def read_passwords(config)
         getent = Yast::Execute.on_target!("/usr/bin/getent", "shadow", stdout: :capture)
-        getent.lines.map do |line|
-          values = line.chomp.split(":")
-          max_age = values[SHADOW_MAPPING["maximum_age"]]
-          inactivity_period = values[SHADOW_MAPPING["inactivity_period"]]
-          expiration = parse_account_expiration(values[SHADOW_MAPPING["account_expiration"]])
-          Password.new(
-            config,
-            values[SHADOW_MAPPING["username"]],
-            value:              values[SHADOW_MAPPING["value"]],
-            last_change:        parse_last_change(values[SHADOW_MAPPING["last_change"]]),
-            minimum_age:        values[SHADOW_MAPPING["minimum_age"]].to_i,
-            maximum_age:        max_age&.to_i,
-            warning_period:     values[SHADOW_MAPPING["warning_period"]].to_i,
-            inactivity_period:  inactivity_period&.to_i,
-            account_expiration: expiration
-          )
+        getent.lines.each do |line|
+          password = parse_getent_password(config, line)
+          user = config.users.find { |u| u.name == password.name }
+          if !user
+            log.warn "Found password for non existing user #{password.name}."
+            next
+          end
+
+          user.password = password
         end
       end
 
+      def parse_getent_password(config, line)
+        values = line.chomp.split(":")
+        max_age = values[SHADOW_MAPPING["maximum_age"]]
+        inactivity_period = values[SHADOW_MAPPING["inactivity_period"]]
+        expiration = parse_account_expiration(values[SHADOW_MAPPING["account_expiration"]])
+        Password.new(
+          config,
+          values[SHADOW_MAPPING["username"]],
+          value:              values[SHADOW_MAPPING["value"]],
+          last_change:        parse_last_change(values[SHADOW_MAPPING["last_change"]]),
+          minimum_age:        values[SHADOW_MAPPING["minimum_age"]].to_i,
+          maximum_age:        max_age&.to_i,
```

Comment:
NP: Why use `values[SHADOW_MAPPING["maximum_age"]]&.to_i` directly here too? The same for the rest.

Reply:
well, reason is different meaning. Maximum age set to 0 means something different then if not set. In minimum age it is same ( disabled, in max age it should not be used as it age immediatelly )

Reply:
Not sure if I got it. Probably I'm missing something, but at first glance both approaches looks the same for me.

I mean, 

```ruby 

max_age = values[SHADOW_MAPPING["maximum_age"]]
something = max_age&.to_i
```

will produce the same result than

```ruby
something = values[SHADOW_MAPPING["maximum_age"]]&.to_i
```

What am I missing?

Reply:
yes, they are same. Just it start to be too long, so use separate value. No other reason.

Reply:
Yes, they looks too long. However, my proposal was just to keep consistence and use the same way to _assign_ the value for all of them (_minimum\_age_ and _maximum\_age_ have equals length). I have no strong opinion about which choose, although I tend to avoid the intermediate variable :)

Anyway, still **LGTM** :stuck_out_tongue_winking_eye: